src/EventLoop.cpp

```diff
@@ -489,11 +481,15 @@ auto requests_accept_async(uv_async_t* handle) -> void
              * immediately.
              */
             request_handle->setCompletionStatus(CURLcode::CURLE_SEND_ERROR);
-            request_handle->onComplete(*event_loop, *request_handle.createSharedRequestOnHeap());
+
+            // If we are calling onComplete now, move the shared_ptr into the method so it cleans itself up correctly.
+            request_handle->onComplete(*event_loop, std::move(*shared_request_on_heap));
```

Comment:
this is still leaking the pointer to the shared_ptr itself. It's not leaking the shared value, but the shared_ptr itself. If the pointer was acquired into a unique_ptr this wouldn't be a problem